LAPACK Required on non-MAC?

The documentation in the README.md file indicates that LAPACK is only required for MAC, but when I run cmake on a Fedora like distribution (Amazon Linux 2 in this case) I get the message.

```
CMake Error at CMakeLists.txt:51 (find_library):
  Could not find LAPACK_LIBRARY using the following names: lapack
```
Looking at the line indicated in the CMakeLists.txt file I don't see any OS based conditions, so if we really don't need it for non-Mac OSes, then we should make the check conditional on the OS.

If we do need it in the non-Mac OSes we need to change the README.md.

--- a/README.md
+++ b/README.md
@@ -74,7 +74,7 @@ brew install cmake openblas eigen libomp lapack
 
 ```bash
 # Install all required packages
-sudo yum install cmake openblas-devel eigen3-devel
+sudo yum install cmake openblas-devel eigen3-devel lapack
 ```
 
 ## Building and Installing

